### ISSUE TITLE
Unix domain socket support

### ISSUE THREAD

**Issue by asauber** [etcd-io/etcd#12450]
Hi. In the [documentation for --listen-client-urls](https://etcd.io/docs/v3.3.12/op-guide/configuration/#--listen-client-urls), the docs say that a unix path like `unix://<file-path>` can be used.

However, due to [this validation code](https://github.com/etcd-io/etcd/blob/v3.4.13/pkg/types/urls.go#L42), unix domain sockets cannot actually be used. We get an error like the following:

```
URL address does not have the form \"host:port\": unix:///var/etcd/data/datafile"
```

Is my use of the flag correct? `--listen-client-urls=unix:///var/etcd/data/datafile`

Has anyone had success using unix sockets with etcd?

**Comment by xiang90** [comment]
the line you mentioned should just verify the host part not the scheme part.

**Comment by justinsb** [comment]
This issue has an open PR, I propose we reopen it.

/reopen

**Comment by HubertZhang** [comment]
Hi I was wondering will this feature be supported on feature release of etcd 3.5? Creating an embedded etcd cluster using unix socket would be very convenient for test purposes😀

**Comment by thequailman** [comment]
Looks like #12469 was merged into main but not backported to 3.5, any chance it could be backported?

**Comment by pohly** [comment]
The backport was merged in https://github.com/etcd-io/etcd/pull/15940.
